component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java

```diff
@@ -0,0 +1,481 @@
+            
+            if (clientId!=null && !clientId.equals(clientID)) {
+                throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR,
+                                         "Token's client ID does not match app's. clientID "
+                                             + "from tokenINFO: " + tokenInfo.get(OAuthConstants.ISSUED_TO_PARAMETER));
+            }
+    
+            if (log.isTraceEnabled()) {
+                log.trace("Successfully validated accessToken from openid : " + tokenInfo);
+            }
+            scopes = tokenInfo.has(OAuthConstants.SCOPE_PARAMETER) ? tokenInfo.getString(OAuthConstants.SCOPE_PARAMETER).split(" ") : null;
+        } catch (JSONException jsonException) {
+            throw new OAuthException(OAuthExceptionCode.ACCESS_TOKEN_ERROR,
+                                     "Error during token validation: token format is ko");
+        }
+    
+        return new OpenIdAccessTokenContext(accessToken.getTokenData(), scopes);
+    
+    }
+    @Override
+    public <C> C getAuthorizedSocialApiObject(OpenIdAccessTokenContext accessToken, Class<C> socialApiObjectType) {
+        return null;
+    }
+    
+    @Override
+    public void saveAccessTokenAttributesToUserProfile(UserProfile userProfile, OAuthCodec codec, OpenIdAccessTokenContext accessToken) {
+        String encodedAccessToken = codec.encodeString(accessToken.accessToken.getAccessToken());
+        OAuthPersistenceUtils.saveLongAttribute(encodedAccessToken, userProfile, OAuthConstants.PROFILE_OPEN_ID_ACCESS_TOKEN,
+                                                false, chunkLength);
+        
+    }
+    
+    @Override
+    public OpenIdAccessTokenContext getAccessTokenFromUserProfile(UserProfile userProfile, OAuthCodec codec) {
+        String encodedAccessToken = OAuthPersistenceUtils.getLongAttribute(userProfile,
+                                                                           OAuthConstants.PROFILE_OPEN_ID_ACCESS_TOKEN, false);
+        String encodedAccessTokenSecret = OAuthPersistenceUtils.getLongAttribute(userProfile, OAuthConstants.PROFILE_LINKEDIN_ACCESS_TOKEN_SECRET, false);
+        String decodedAccessToken = codec.decodeString(encodedAccessToken);
+        String decodedAccessTokenSecret = codec.decodeString(encodedAccessTokenSecret);
+        
+        if(decodedAccessToken == null || decodedAccessTokenSecret == null) {
+            return null;
+        } else {
+            OAuth2AccessToken token = new OAuth2AccessToken(decodedAccessToken, decodedAccessTokenSecret);
+            return new OpenIdAccessTokenContext(token);
+        }
+    }
+    
+    @Override
+    public void removeAccessTokenFromUserProfile(UserProfile userProfile) {
+        OAuthPersistenceUtils.removeLongAttribute(userProfile, OAuthConstants.PROFILE_LINKEDIN_ACCESS_TOKEN, false);
+        OAuthPersistenceUtils.removeLongAttribute(userProfile, OAuthConstants.PROFILE_LINKEDIN_ACCESS_TOKEN_SECRET, false);
+    }
```

Comment:
IMO PROFILE_OPEN_ID instead of PROFILE_LINKEDIN

Reply:
done